api/main.py

```diff
@@ -269,6 +269,19 @@ async def get_root_node(node_id: str):
 async def post_node(node: Node, token: str = Depends(get_user)):
     """Create a new node"""
     try:
+        if node.parent:
+            parent = await db.find_by_id(Node, node.parent)
+            if not parent:
+                raise HTTPException(
+                    status_code=status.HTTP_400_BAD_REQUEST,
```

Comment:
I think it should rather get HTTP `404` status than `400`. Is there a reason we want to return `Bad Request` here?